hw/ip/otbn/dv/otbnsim/sim/insn.py

```diff
@@ -291,10 +291,7 @@ def __init__(self, op_vals: Dict[str, int]):
         self.grs1 = op_vals['grs1']
 
     def execute(self, state: OTBNState) -> None:
-        if self.grd == 0:
-            return
-
-        old_val = state.read_csr(self.csr)
+        old_val = state.read_csr(self.csr) if self.grd != 0 else 0
         new_val = state.gprs.get_reg(self.grs1).read_unsigned()
 
         state.gprs.get_reg(self.grd).write_unsigned(old_val)
```

Comment:
This line is unnecessary for x0 as well.

I'd align the code in this execute function more closely with the spec, which has it as

```
gpr_s_val = GPR[s]

if d != 0:
  GPR[d] = CSR[csr_num]

CSR[csr_num] = gpr_s_val

```

Reply:
I wrote it like that to avoid reading the CSR if `grd` was `x0`, but your suggestion is much cleaner! I'll push up a version like that.